src/main/java/hachi/main/Parser.java

```diff
@@ -0,0 +1,54 @@
+package hachi.main;
+
+import hachi.commands.*;
+
+/**
+ * Handles user instruction and perform the corresponding type of command.
+ */
+public class Parser {
+    private enum Type {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    /**
+     *
+     * @param input The user's input string.
+     * @return A specific type of command.
+     * @throws IllegalArgumentException if command is unknown.
+     */
+    public static Command parse(String input) throws IllegalArgumentException {
+        String[] words = input.split(" ");
+        Type t = Type.valueOf(words[0].toUpperCase());
+        switch (t) {
```

Comment:
Perhaps indentation for case: can be removed to follow the coding standard. But other than that, your switch/case looks damn nice and neat :)

Reply:
Thank you for pointing that out!